fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/SimplifyComparisonPredicate.java

```diff
@@ -111,13 +113,56 @@ private Expression processDateLikeTypeCoercion(ComparisonPredicate cp, Expressio
             }
         }
 
+        if (left.getDataType() == DateType.INSTANCE && right.getDataType() == DateType.INSTANCE) {
```

Comment:
do storage engine support date slot compare with datetime?
if it support, we could avoid add cast on the slot
and then, this expression could push down to storage engine.